testing.md

```diff
@@ -179,3 +179,92 @@ For example integration see:
 - https://deno.land/std@$STD_VERSION/testing/chai_example.ts
 - https://deno.land/std@$STD_VERSION/testing/sinon_example.ts
 - https://deno.land/std@$STD_VERSION/testing/fast_check_example.ts
+
+### Example: spying on a function with Sinon
+
+Test spies are function stand-ins that are used to assert if a function's
+internal behavior matches expectations. Sinon is a widely used testing library
+that provides test spies and can be used in Deno by importing it from a CDN,
+such as Skypack:
+
+```js
+import sinon from "https://cdn.skypack.dev/sinon";
+```
+
+Consider the following functions `foo` and `bar`, where `foo` makes a call to
+`bar`:
+
+```js
+function bar() {/*...*/}
+
+function foo() {
+  bar();
+}
+```
+
+Say we want to assert that `bar` is called during execution of `foo`. There are
+a few ways to achieve this with Sinon, one is to have function `foo` take a
+parameter and pass `bar`:
+
+```js
+function foo(bar) {
+  bar();
+}
+```
+
+This way, we can create a wrapper around `bar` with sinon to create a test spy:
+
+```js
+import sinon from "https://cdn.skypack.dev/sinon";
+import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+import { bar, foo } from "./my_file.js";
```

Comment:
Would be great to have a concrete `my_file.js` defined on this page - it's not clear this of examples above to use

Reply:
you mean using some more concrete functions instead of `foo` and `bar`?

Reply:
Using `foo` and `bar` is fine, I mean that there should be a concrete `my_file.js` example, because right now there are numerous codeblocks defining these functions.

Reply:
Okay, I made some improvements (I hope)